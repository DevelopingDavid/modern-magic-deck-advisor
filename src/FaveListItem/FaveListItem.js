import React, { Component } from "react";
import "./FaveListItem.scss";

class FaveListItem extends Component {
  removeFaveListItem = event => {
    // this.props.removeFaveListItem(event.target.closest("li").id)
  }

  render() {
    return (
      <li id={this.props.deckIndex}>
        {this.props.deckName}
        <i 
          onClick={this.removeFaveListItem}
          className="far fa-trash-alt" />
      </li>
    )
  }
}

export default FaveListItem;