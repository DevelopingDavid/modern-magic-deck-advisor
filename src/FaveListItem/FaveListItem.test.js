import React from 'react';
import FaveListItem from './FaveListItem';
import { shallow } from 'enzyme';
import { isMainThread } from 'worker_threads';

const deckName = "Bant Spirits"
const index = 0;
const userFavoriteDecks = [
  {
    deckName: "Bant Spirits",
    cards: [
    "Breeding Pool",
    "Cavern of Souls",
    "Flooded Strand",
    "Horizon Canopy",
    "Misty Rainforest",
    "Windswept Heath",
    "Noble Hierarch",
    "Phantasmal Image",
    "Collected Company",
    "Aether Vial"
    ],
    cardCounts: {
    Breeding Pool: 1,
    Cavern of Souls: 1,
    Flooded Strand: 3,
    Horizon Canopy: 3,
    Misty Rainforest: 2,
    Windswept Heath: 2,
    Noble Hierarch: 4,
    Phantasmal Image: 3,
    Collected Company: 4,
    Aether Vial: 3
    }
    }
]

describe("FaveListItem", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FaveListItem />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})