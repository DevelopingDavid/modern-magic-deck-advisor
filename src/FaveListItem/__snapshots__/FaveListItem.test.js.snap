// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaveListItem should match the snapshot 1`] = `
<li
  className="list--li"
  id={0}
>
  Bant Spirits
  <i
    className="far fa-trash-alt"
    onClick={[Function]}
  />
</li>
`;
